pkgs/development/python-modules/pyobjc-framework-CoreBluetooth/default.nix

```diff
@@ -0,0 +1,66 @@
+{
+  buildPythonPackage,
+  darwin,
+  fetchFromGitHub,
+  lib,
+  pyobjc-core,
+  pyobjc-framework-Cocoa,
+  setuptools,
+}:
+
+buildPythonPackage rec {
+  pname = "pyobjc-framework-CoreBluetooth";
+  version = "11.1";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "ronaldoussoren";
+    repo = "pyobjc";
+    tag = "v${version}";
+    hash = "sha256-2qPGJ/1hXf3k8AqVLr02fVIM9ziVG9NMrm3hN1de1Us=";
+  };
+
+  sourceRoot = "${src.name}/pyobjc-framework-CoreBluetooth";
+
+  build-system = [ setuptools ];
+
+  buildInputs = [
+    darwin.libffi
+  ];
+
+  nativeBuildInputs = [
+    darwin.DarwinTools # sw_vers
+  ];
+
+  # See https://github.com/ronaldoussoren/pyobjc/pull/641. Unfortunately, we
+  # cannot just pull that diff with fetchpatch due to https://discourse.nixos.org/t/how-to-apply-patches-with-sourceroot/59727.
```

Comment:
Can't you use `cd` in `postPatch`?
However, don't use the linked PR but rather the commits that made it upstream.